error: redundant redeclaration of 'cv::String cv::format(const char*, ...)'

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => (master) heads/master-0-g090ee46
- Operating System / Platform => Linux
- Compiler => GCC/G++

##### Detailed description

Programs fail to compile when `-Werror -Wredundant-decls` GCC compile options are used and when including `<opencv2/core.hpp>`

##### Steps to reproduce

<!-- to add code example fence it with triple backticks and optional file extension
    ```.cpp
    // C++ code example
    ```
 or attach as .txt or .zip file
-->
```.cpp
// file test.cpp
#include <opencv2/core.hpp>

int main(int argc, char** argv)
{
  return 0;
}
```
```
$ g++ -Wredundant-decls -Werror `pkg-config --cflags --libs opencv` test.cpp

In file included from /opt/media/install/include/opencv2/core.hpp:3261:0,
                 from test.cpp:1:
/opt/media/install/include/opencv2/core/utility.hpp:191:48: error: redundant redeclaration of 'cv::String cv::format(const char*, ...)' in same scope [-Werror=redundant-decls]
 CV_EXPORTS String format( const char* fmt, ... );
                                                ^
In file included from /opt/media/install/include/opencv2/core.hpp:3259:0,
                 from test.cpp:1:
/opt/media/install/include/opencv2/core/operations.hpp:368:19: error: previous declaration of 'cv::String cv::format(const char*, ...)' [-Werror=redundant-decls]
 CV_EXPORTS String format( const char* fmt, ... );
                   ^
cc1plus: all warnings being treated as errors
```

--- a/modules/core/include/opencv2/core/operations.hpp
+++ b/modules/core/include/opencv2/core/operations.hpp
@@ -365,6 +365,12 @@ template<typename _Tp> static inline _Tp randu()
 
 ///////////////////////////////// Formatted string generation /////////////////////////////////
 
+/** @brief Returns a text string formatted using the printf-like expression.
+
+The function acts like sprintf but forms and returns an STL string. It can be used to form an error
+message in the Exception constructor.
+@param fmt printf-compatible formatting specifiers.
+ */
 CV_EXPORTS String format( const char* fmt, ... );
 
 ///////////////////////////////// Formatted output of cv::Mat /////////////////////////////////

--- a/modules/core/include/opencv2/core/utility.hpp
+++ b/modules/core/include/opencv2/core/utility.hpp
@@ -182,13 +182,6 @@ extern "C" typedef int (*ErrorCallback)( int status, const char* func_name,
 */
 CV_EXPORTS ErrorCallback redirectError( ErrorCallback errCallback, void* userdata=0, void** prevUserdata=0);
 
-/** @brief Returns a text string formatted using the printf-like expression.
-
-The function acts like sprintf but forms and returns an STL string. It can be used to form an error
-message in the Exception constructor.
-@param fmt printf-compatible formatting specifiers.
- */
-CV_EXPORTS String format( const char* fmt, ... );
 CV_EXPORTS String tempfile( const char* suffix = 0);
 CV_EXPORTS void glob(String pattern, std::vector<String>& result, bool recursive = false);
 

